test/TestChannelFactory.js

```diff
@@ -1,287 +1,293 @@
-
-        });
-
-        describe("Channel validation", function () {
-
-            var channelfactory = new Factory({validate: true});
-
-            it("factory passes on validation config", function () {
-                assert.isTrue(channelfactory.coreapi.isValidating());
-            });
-
-            it("CoreApi uses default resolver from ChannelFactory", function () {
-                var channel = channelfactory.get(MockChannel);
-                assert.equal(channel.channelName, MockChannel.id);
-            });
-
-        });
-
-    });
-
+define([
+    "bullhorn/ChannelFactory",
+    "./MockChannel"
+], function (
+    Factory,
+    MockChannel
+) {
+
+    "use strict";
+
+    describe("Test ChannelFactory, ensuring that it correctly wraps CoreApi methods", function () {
+
+        var callSucceeded = false,
+            callSucceeded2 = false,
+            receiveFunc = function () { callSucceeded = true; },
+            receiveFunc2 = function () { callSucceeded2 = true; },
+            testScope1 = {scope: 1, name: "scope1"},
+            testScope2 = {scope: 2, name: "scope2"},
+            message = {data: 'test'};
+
+        beforeEach(function () {
+            callSucceeded = false;
+            callSucceeded2 = false;
+        });
+
+        describe("Get channel from factory", function () {
+
+            var resolverExecuted = false,
+                resolver = function () {
+                    resolverExecuted = true;
+                    return MockChannel;
+                },
+                channelfactory = new Factory({resolver: resolver});
+
+            beforeEach(function () {
+                resolverExecuted = false;
+            });
+
+            it("get using string name uses name resolver function", function () {
+                channelfactory.get("test/MockChannel", testScope1);
+                assert.isTrue(resolverExecuted);
+            });
+
+            it("get using string name returns correct channel from factory", function () {
+                var channel = channelfactory.get("test/MockChannel", testScope1);
+                assert.equal(channel.channelName, MockChannel.id);
+            });
+
+            it("get using channel object uses default object resolver function", function () {
+                channelfactory.get(MockChannel, testScope1);
+                assert.isFalse(resolverExecuted);
+            });
+
+            it("get using channel object returns correct channel from factory", function () {
+                var channel = channelfactory.get(MockChannel, testScope1);
+                assert.equal(channel.channelName, MockChannel.id);
+            });
+            
+            it("get using a string name returns the default channel from factory", function () {
+                var channelfactory = new Factory({});
+                var channel = channelfactory.get("test/DefaultChannel", testScope1);
+                assert.equal(channel.channelName, "default/DefaultChannel");

```

Comment:
With the change suggested above to use the name as the id, this assertion should match. E.g.:
```
var channel = channelfactory.get("any-name-i-choose", testScope1);
assert.equal(channel.channelName, "any-name-i-choose");
```